tests/test_client_type_errors.py

```diff
@@ -0,0 +1,245 @@
+"""
+This file exists to test for type-checker false positives and false negatives.
+It doesn't contain any test functions.
+"""
+
+from dataclasses import dataclass
+from unittest.mock import Mock
+
+from temporalio import workflow
+from temporalio.client import (
+    Client,
+    WithStartWorkflowOperation,
+    WorkflowHandle,
+    WorkflowUpdateHandle,
+    WorkflowUpdateStage,
+)
+from temporalio.common import WorkflowIDConflictPolicy
+from temporalio.service import ServiceClient
+
+
+@dataclass
+class WorkflowInput:
+    pass
+
+
+@dataclass
+class SignalInput:
+    pass
+
+
+@dataclass
+class QueryInput:
+    pass
+
+
+@dataclass
+class UpdateInput:
+    pass
+
+
+@dataclass
+class WorkflowOutput:
+    pass
+
+
+@dataclass
+class QueryOutput:
+    pass
+
+
+@dataclass
+class UpdateOutput:
+    pass
+
+
+@workflow.defn
+class TestWorkflow:
+    @workflow.run
+    async def run(self, _: WorkflowInput) -> WorkflowOutput:
+        return WorkflowOutput()
+
+    @workflow.signal
+    async def signal(self, _: SignalInput) -> None:
+        pass
+
+    @workflow.query
+    async def query(self, _: QueryInput) -> QueryOutput:
+        return QueryOutput()
+
+    @workflow.update
+    async def update(self, _: UpdateInput) -> UpdateOutput:
+        return UpdateOutput()
+
+
+@workflow.defn
+class TestWorkflow2(TestWorkflow):
+    @workflow.run
+    async def run(self, _: WorkflowInput) -> WorkflowOutput:
+        return WorkflowOutput()
+
+
+async def _test_start_and_execute_workflow_type_errors():
```

Comment:
Pardon my lack of understanding of this new type checking stuff mixed in with regular tests. Just browsing the code, it is hard to understand when this code is executed or how it works compared to other tests. Does it run or is it just type-checked? If not, I wonder if a decorator or some other name would be clearer for "test that does not run" rather than just `_test` prefix.